### DISCUSSION TITLE
Why are backward and forward mode shapes different when there is no rotation

### DISCUSSION THREAD

**Discussion by wigging** [petrobras/ross#963] [Q&A]
This example is for a solid steel shaft with an undamped bearing at each end of the shaft. Notice that for the modal analysis, there is no rotation speed so `rotor.run_modal(speed=0)`.

```python
import matplotlib.pyplot as plt
import ross as rs

# Steel material properties
steel = rs.Material(name='Steel', rho=7810, E=211e9, G_s=81.2e9)

# Number of shaft elements
N_elem = 6

# Dimensions of a shaft element
L_total = 2.032
L = L_total / N_elem
i_d = 0
o_d = 0.127

shaft_elem = rs.ShaftElement(
    L=L,
    idl=i_d,
    odl=o_d,
    material=steel,
    shear_effects=True,
    rotary_inertia=True,
    gyroscopic=True
)

shaft_elements = [shaft_elem for _ in range(N_elem)]

# Bearings at beginning and end of shaft
kb = 175126834.64
bearing0 = rs.BearingElement(n=0, kxx=kb, cxx=0)
bearing1 = rs.BearingElement(n=N_elem, kxx=kb, cxx=0)
bearings = [bearing0, bearing1]

# Rotor assembly
rotor = rs.Rotor(shaft_elements=shaft_elements, bearing_elements=bearings)

# Run modal analysis and store results
results = rotor.run_modal(speed=0)

# Plot results using Matplotlib

fig1, axs1 = plt.subplots(nrows=1, ncols=3, sharey=True, figsize=(8, 4.8), tight_layout=True)
fig2, axs2 = plt.subplots(nrows=1, ncols=3, sharey=True, figsize=(8, 4.8), tight_layout=True)

for i in range(N_elem):

    fig_plotly = results.shapes[i].plot_2d(orientation='y')
    x = fig_plotly.data[0]['x']
    y = fig_plotly.data[0]['y']

    k = i // 2

    if i % 2 == 0:
        # Plot mode shapes for backward rotation
        axs1[k].plot(x, y)
        axs1[k].set_ylim(-1.2, 1.2)
        axs1[k].set_xlabel('Rotor length [m]')
        axs1[k].set_title(f'Mode {i}')
        if k == 0:
            axs1[k].set_ylabel('Relative displacement [-]')
    else:
        # Plot mode shapes for forward rotation
        axs2[k].plot(x, y)
        axs2[k].set_ylim(-1.2, 1.2)
        axs2[k].set_xlabel('Rotor length [m]')
        axs2[k].set_title(f'Mode {i}')
        if k == 0:
            axs2[k].set_ylabel('Relative displacement [-]')

plt.show()
```

Running the example gives the following mode shapes for backward rotation:
![Figure_1](https://user-images.githubusercontent.com/6828967/226949975-54a0550b-a86a-4b4a-bd2f-ea899fc8b037.png)

And running the example gives the following mode shapes for forward rotation:
![Figure_2](https://user-images.githubusercontent.com/6828967/226950009-2dd1a010-d795-443b-915c-cdba78e00ab6.png)

I think modes 0, 2, and 4 represent the 1st, 2nd, and 3rd mode shapes for backward rotation. And modes 1, 3, and 5 are the 1st, 2nd, and 3rd mode shapes for forward rotation. Is that correct? 

If my understanding is correct, then why are the 2nd mode shapes (mode 2 and mode 3) not the same? I would expect them to be the same since there is no shaft rotation. Or am I plotting the wrong data?

**Comment by raphaeltimbo** [comment]
Hi @wigging !
Since mode shapes are just eigenvectors of a matrix, if you have a scalar multiplying that eigenvector you still have the same eigenvector. 
Therefore, you are actually getting correct mode shapes, they are just in a different scale. 
To make it easier to compare this mode shapes you could normalize them. One possible normalization is to make the largest relative displacement be equal to one. You can accomplish that adding this to you code:

```python
...
    fig_plotly = results.shapes[i].plot_2d(orientation='y')
    x = fig_plotly.data[0]['x']
    y = fig_plotly.data[0]['y']
    # normalize y values
    iymax = np.argmax(abs(y))
    y = y / y[iymax]
...
```

**Reply by wigging** [reply]
For the `ModalResults`, it looks like the data is already normalized. See line [855](https://github.com/petrobras/ross/blob/ac9f339e5939365f6907b1abc9fc567f35247e0d/ross/results.py#L855) in the `ModalResults` class.

**Reply by wigging** [reply]
I implemented your suggestion to normalize the y values and now the shapes are the same for the backward and forward rotations. But now I'm confused about why the `Shape` argument is already set to `normalize=True` as mentioned in my previous comment. It looks like the normalize step in `Shape` will normalize the `evec` or eigenvector. Does this not make the x and y values from `x = fig_plotly.data[0]['x']` and `y = fig_plotly.data[0]['y']` normalized too?

**Reply by raphaeltimbo** [reply]
Hi @wigging !
The normalization that we apply now won't make both the max x and max y value be equal to 1. Normalizing like that is actually not possible if we have nodes with elliptical orbits.
In fact, I think we should actually change the code to normalize so that the max major axis is equal to 1.
